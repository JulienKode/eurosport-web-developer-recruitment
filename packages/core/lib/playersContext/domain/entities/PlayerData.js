'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
var PlayerDataKeys;
(function(PlayerDataKeys) {
  PlayerDataKeys['rank'] = 'rank';
  PlayerDataKeys['points'] = 'points';
  PlayerDataKeys['weight'] = 'weight';
  PlayerDataKeys['height'] = 'height';
  PlayerDataKeys['age'] = 'age';
  PlayerDataKeys['last'] = 'last';
})(PlayerDataKeys || (PlayerDataKeys = {}));
