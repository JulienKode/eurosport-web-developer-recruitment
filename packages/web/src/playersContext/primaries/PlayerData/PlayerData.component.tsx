import React from 'react';
import { PlayerData } from '@app/core/src/playersContext/domain/entities/PlayerData';
import {
  EWeightUnit,
  formatGrams,
} from '@app/core/src/common/formatters/weight';
import { Text } from '@app/web/src/components/Text/Text.component';
import styled from 'styled-components';

export const PlayerDataComponent = ({
  points,
  rank,
  age,
  weight,
  height,
}: PlayerData) => {
  const playerRank = `Rang: ${rank}`;
  const playerPoint = `Point: ${points}`;
  const playerAge = `Age: ${age} ans`;
  const formattedWeight = formatGrams(weight, EWeightUnit.kilograms);
  const playerWeight = `Poids: ${formattedWeight} kg`;
  const playerHeight = `Taille: ${height} cm`;
  return (
    <PlayerDataContainer>
      <Text>{playerRank}</Text>
      <Text>{playerPoint}</Text>
      <Text>{playerAge}</Text>
      <Text>{playerWeight}</Text>
      <Text>{playerHeight}</Text>
    </PlayerDataContainer>
  );
};

const PlayerDataContainer = styled.div`
display: flex;
flex 1;
justify-content: center;
flex-direction: column;
`;
