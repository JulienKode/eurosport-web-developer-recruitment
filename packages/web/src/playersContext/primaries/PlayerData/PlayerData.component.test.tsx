import React from 'react';
import { shallow } from 'enzyme';
import { PlayerDataComponent } from './PlayerData.component';

describe('PlayerDataComponent component', () => {
  it('should match snapshot', () => {
    // Given
    const wrapper = shallow(
      <PlayerDataComponent
        rank={1}
        points={1000}
        weight={72000}
        height={185}
        age={22}
        last={[1, 1, 1, 1, 1]}
      />,
    );

    // Expect
    expect(wrapper).toMatchSnapshot();
  });
});
