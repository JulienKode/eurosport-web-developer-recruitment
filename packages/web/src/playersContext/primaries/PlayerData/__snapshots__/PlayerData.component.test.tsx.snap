// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoRequests component should match snapshot 1`] = `ShallowWrapper {}`;
